cloudv2/counter.w

```diff
@@ -0,0 +1,237 @@
+bring aws;
+bring cloud;
+bring fs;
+bring ui;
+bring util;
+bring sim;
+
+bring "@cdktf/provider-aws" as tfaws;
+bring "./util.w" as myutil;
+
+// TODO: Default values for struct fields - https://github.com/winglang/wing/issues/3121
+
+pub struct CounterProps {
+  /// The initial value of the counter
+  /// @default 0
+  initial: num?;
+}
+
+interface ICounter {
+  /// Increments the counter atomically by a certain amount and returns the previous value.
+  /// - `amount` The amount to increment by (defaults to 1)
+  /// - `key` The key of the counter (defaults to "default")
+  inflight inc(amount: num?, key: str?): num;
+
+  /// Decrements the counter atomically by a certain amount and returns the previous value.
+  /// - `amount` The amount to decrement by (defaults to 1)
+  /// - `key` The key of the counter (defaults to "default")
+  inflight dec(amount: num?, key: str?): num;
+
+  /// Returns the current value of the counter.
+  /// - `key` The key of the counter (defaults to "default")
+  inflight peek(key: str?): num;
+  
+  /// Sets the value of the counter.
+  /// - `value` The new value of the counter
+  /// - `key` The key of the counter (defaults to "default")
+  inflight set(value: num, key: str?): void;
+}
+
+pub class Counter impl ICounter {
+  inner: ICounter;
+  pub initial: num;
+  new(props: CounterProps) {
+    nodeof(this).title = "Counter";
+    nodeof(this).description = "A distributed atomic counter";
+    nodeof(this).icon = "calculator";
+    nodeof(this).color = "lime";
+
+    let target = util.env("WING_TARGET");
+    let id = nodeof(this).id;
+    this.initial = props.initial ?? 0;
+    if target == "sim" {
+      this.inner = new Counter_sim(props) as id;
+    } elif target == "tf-aws" {
+      this.inner = new Counter_tfaws(props) as id;
+    } else {
+      throw "Unsupported target: " + target;
+    }
```

Comment:
This means that I cannot override the implementation with a platform extension, correct?

I suspect we need to address this before we finish the migration because we have users who rely on this capability.

Reply:
Yap